lib/internal/modules/esm/loader.js

```diff
@@ -70,27 +102,35 @@ class ESMLoader {
   /**
    * Prior to ESM loading. These are called once before any modules are started.
    * @private
-   * @property {Function[]} globalPreloaders First-in-first-out list of
-   * preload hooks.
+   * @property {KeyedHook[]} globalPreloaders Last-in-first-out
+   *  list of preload hooks.
    */
   #globalPreloaders = [];
 
   /**
    * Phase 2 of 2 in ESM loading.
    * @private
-   * @property {Function[]} loaders First-in-first-out list of loader hooks.
+   * @property {KeyedHook[]} loaders Last-in-first-out
+   *  collection of loader hooks.
    */
   #loaders = [
-    defaultLoad,
+    {
+      fn: defaultLoad,
+      url: 'node:lib/internal/modules/esm/load.js',
```

Comment:
@aduh95 Is this a valid use of the `node:` protocol? If not, is there some other way to represent something like `node-internals:lib/internal/modules/esm/load.js`?

Reply:
I guess the actual correct URL would not use the file extension and not include the `lib/` part.
```suggestion
      url: 'node:internal/modules/esm/load',
```

Reply:
But this is an ESM context, so why wouldn’t it include the extension?

Reply:
That's how URL of core modules work:

```
node --input-type=module -e 'throw new Error'
file:///Users/duhamean/Documents/node/[eval1]:1
throw new Error
      ^

Error
    at file:///Users/duhamean/Documents/node/[eval1]:1:7
    at ModuleJob.run (node:internal/modules/esm/module_job:198:25)
    at async ESMLoader.eval (node:internal/modules/esm/loader:252:24)
    at async loadESM (node:internal/process/esm_loader:85:5)
    at async handleMainPromise (node:internal/modules/run_main:61:12)

Node.js v18.0.0
```

See in the stack trace, no file extensions, no `lib/` either.

Reply:
That’s probably legacy from the CommonJS loader that we haven’t updated. I would think that nowadays this stack trace _should_ include the extensions, for both loaders. It could be its own PR, though.

Reply:
I would lean towards including the file extension, but perhaps for now, exclude it for consistency and add it back as part of a larger change to the others?

Reply:
Why is the file extension important for something that will never exist as an editable file on disk for node users?

Reply:
> for now, exclude it for consistency and add it back as part of a larger change to the others?

Yes, we should be consistent and make this change in its own PR. It'll probably cause a lot of tests to need updating.